src/datasets/utils/spark_utils.py

```diff
@@ -0,0 +1,48 @@
+import os
+import uuid
+from typing import Optional
+
+import pyspark
```

Comment:
We need to put this import into try catch. pyspark is not a hard dependencies for datasets.